web/controllers/v1/user_controller.ex

```diff
@@ -56,4 +58,29 @@ defmodule Cog.V1.UserController do
     Repo.delete!(user)
     send_resp(conn, :no_content, "")
   end
+
+  ###############################################
+  # Plug function - local only to user controller
+  ###############################################
+  @spec check_self_updating(Plug.Conn.t, []) :: Plug.Conn.t
+  def check_self_updating(conn, opts \\ []) do
+    if determine_self_updating(conn) do
+      conn
+    else
+      plug_opts = Cog.Plug.Authorization.init(opts)
+      Cog.Plug.Authorization.call(conn, plug_opts)
+    end
+  end
+
+    @doc """
+  Store the self updating user flag in the `conn`.
+  """
+  @spec determine_self_updating(%Plug.Conn{}) :: true | false
+  def determine_self_updating(conn) do
+    Map.has_key?(conn.private, :phoenix_action) and
```

Comment:
I thought so too, but during the research phase, I found there were times when it was not present. It may be always at this stage. I'll verify.